### PR TITLE
simpleiot-js: Update to new NATS API

### PR DESCRIPTION
**Pull request by bminer**
Also added tests, linting, and updated .prettierrc.

This is to be merged into PR #448.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,2 +1,7 @@
-printWidth: 80
```
Comment (cbrake) [inline .prettierrc:1]: Can we keep printWidth? it makes Markdown look nicer.

**Review by cbrake — COMMENTED** [review]
@bminer could we add a function to envsetup.sh to run the frontend lib tests?

File: .prettierrc (hunk 1 above)
Comment (bminer) [reply .prettierrc:1]: I think 80 is the default for Prettier. I don't think this line is strictly needed.

**Comment by bminer** [comment]
> @bminer could we add a function to envsetup.sh to run the frontend lib tests?

Yes. Great idea! I'd presume this function run the SIOT server in the background, as well? And, any ideas on how to tell the store to "reset" itself? I'd like to implement that CLI flag.

File: .prettierrc (hunk 1 above)
Comment (cbrake) [reply .prettierrc:1]: Great, we'll run with it as then.

**Comment by cbrake** [comment]
we could pass a flag to the store component that remove db files before starting up, or we could drop all tables on startup:

https://stackoverflow.com/questions/38672579/delete-all-tables-from-sqlite-database

I kind of like drop all tables, this way if other applications had the DB open, things would not get messed up.

We should also use a different db name than the default.

**Comment by cbrake** [comment]
I added a dblab function to envsetup.sh:

```
siot_dblab() {
  go run github.com/danvergara/dblab@latest --db siot.sqlite --driver sqlite3
}
```

Ideally, I'd like to get the frontend test to run as part of Gitlab CI.

**Comment by bminer** [comment]
Latest commit 717cc9bb7d0bb696ccca5c7c1544018c9e56789d adds `--resetStore` CLI flag to delete all data from `"meta", "edges", "node_points", "edge_points"` tables at start-up.

**Pull request merged**